pandas/core/construction.py

```diff
@@ -499,6 +499,9 @@ def sanitize_array(
     if isinstance(data, ma.MaskedArray):
         data = sanitize_masked_array(data)
 
+    if isinstance(data, np.matrix):
```

Comment:
at the very least needs to be an elif. this is going to slow down everything (though it maybe immaterial).